pkgs/applications/graphics/imv/default.nix

```diff
@@ -1,61 +1,116 @@
-{ asciidoc
+{ stdenv
+, lib
+, fetchFromGitHub
+, asciidoc
 , cmocka
 , docbook_xsl
-, fetchFromGitHub
+, libxslt
 , fontconfig
-, freeimage
+, meson
+, ninja
+, pkgconfig
 , icu
+, pango
+, inih
+, withWindowSystem ? "all"
+, xorg
+, libxkbcommon
 , libGLU
-, libheif
+, wayland
+, withFreeimage ? true
+, freeimage
+, withLibtiff ? true
+, libtiff
```

Comment:
I'm not sure if this practice has changed, but the pattern used to be
```
, withSomething ? true
, libsomething ? null
...
}:

assert withSomething -> libsomething != null;
```
so that the package can be called without `libsomething`.

Reply:
That's not necessary since `nix` is evaluated lazily. The resulting derivation won't reference `libsomething` and if you absolutely want to you can do the following without an evaluation error:

```nix
imv.override {
  withFreeimage = false;
  freeimage = null;
}
```

Shouldn't be necessary though. This approach saves a lot of asserts in our case and makes the code simpler and easier to read and I don't know of any disadvantages.